k8s/base.py

```diff
@@ -78,11 +79,18 @@ def list(cls, namespace="default"):
         return [cls.from_dict(item) for item in resp.json()[u"items"]]
 
     @classmethod
-    def watch_list(cls):
+    def watch_list(cls, namespace=None):
         """Return a generator that yields WatchEvents of cls"""
-        url = cls._meta.watch_list_url
-        if not url:
-            raise NotImplementedError("Cannot watch_list, no watch_list_url defined on class {}".format(cls))
+        if namespace:
+            if cls._meta.watch_list_url_template:
+                url = cls._meta.watch_list_url_template.format(namespace=namespace)
+            else:
+                raise NotImplementedError("Cannot watch_list, no watch_list_url_template defined on class {}".format(cls))
```

Comment:
Maybe we should say "Cannot watch_list with namespace" when you use the namespace parameter, to make it clear why this is different from when `watch_list_url` is not defined?

Reply:
Yeah I agree, I'll fix that